### PR TITLE
Log environment-derived properties per dotnet/msbuild#7484

### PR DESCRIPTION
**Pull request by Forgind**
We will need to update MSBuildPackageVersion once dotnet/msbuild#7484 merges.

### CODE REVIEW

File: src/StructuredLogger/Construction/Construction.cs [hunk 1]
```diff
@@ -80,8 +80,15 @@ public void BuildStarted(object sender, BuildStartedEventArgs args)
                 lock (syncLock)
                 {
                     Build.StartTime = args.Timestamp;
-                    var properties = Build.GetOrCreateNodeWithName<Folder>(Intern(Strings.Environment));
-                    AddProperties(properties, args.BuildEnvironment);
+                    if (args.BuildEnvironment.Count > 0)
```
Comment (KirillOsenkov) [inline src/StructuredLogger/Construction/Construction.cs:83]: need to check for null just in case

**Review by Forgind — COMMENTED** [review]
We should be able to revert everything related to BuildFinishedEventArgs2 as well as the "EnvironmentDerivedProperties" string.

File: src/StructuredLogger/BinaryLogger/BinaryLogger.cs [hunk 2]
```diff
@@ -55,7 +55,7 @@ public sealed class BinaryLogger : ILogger
         //   - TargetSkippedEventArgs: added OriginallySucceeded, Condition, EvaluatedCondition
         // version 14:
         //   - TargetSkippedEventArgs: added SkipReason, OriginalBuildEventContext
-        internal const int FileFormatVersion = 14;
+        internal const int FileFormatVersion = 15;
```
Comment (KirillOsenkov) [inline src/StructuredLogger/BinaryLogger/BinaryLogger.cs:58]: no longer need to rev the format I think

File: src/StructuredLogViewer/Entrypoint.cs [hunk 3]
```diff
@@ -10,6 +11,7 @@ public class Entrypoint
         [STAThread]
         public static void Main(string[] args)
         {
+            Debugger.Launch();
```
Comment (KirillOsenkov) [inline src/StructuredLogViewer/Entrypoint.cs:14]: probably don't want this change

Comment (Forgind) [reply src/StructuredLogViewer/Entrypoint.cs:14]: We don't want every MSBuildStructuredLogViewer user to try to debug it?? How else do we form the next generation of MSBuild devs? /s

**Comment by KirillOsenkov** [comment]
Thanks!

**Pull request merged**